src/aap_eda/core/models/mixins.py

```diff
@@ -0,0 +1,25 @@
+from django.core.exceptions import FieldDoesNotExist
+from django.utils import timezone
+
+
+class ModifiedAtUpdaterMixin:
+    """Mixin to ensure that the `modified_at` field is updated.
+
+    Ensure that the `modified_at` field is updated when the model
+    is saved with specific fields (`update_fields`) specified for update.
+    Only works for `auto_now` fields.
+    """
+
+    def save(self, *args, **kwargs):
+        try:
+            field = self._meta.get_field("modified_at")
+        except FieldDoesNotExist:
+            field = None
+        if (
+            field is not None
+            and hasattr(field, "auto_now")
+            and field.auto_now
+            and kwargs.get("update_fields", [])
+        ):
+            self.modified_at = timezone.now()
```

Comment:
Would kwargs["update_fields"].append("modified_at") work? It should automatically populate the timestamp.
The concern is the in memory object may not have the latest modified_at value after the save.

I am surprised to see you only set the new value for `modified_at` but not include this field in `update_fields` array and the new value is still saved. How can it work?

Reply:
It doesn't. I forgot to add that line. The test is wrong, I need to refresh the object. Thanks, very good catch!